lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -831,26 +830,38 @@ def make_mpiexec_hydra_options(self):
             else:
                 self.log.debug("make_mpiexec_hydra_options: no rmk from HYDRA_RMK %s and hydra_info %s",
                                self.HYDRA_RMK, self.hydra_info)
+
+        default_launcher = getattr(self, 'HYDRA_LAUNCHER', None)
+        if self.options.launcher:
+            launcher = [self.options.launcher]
+            launcher.extend(default_launcher)
         else:
-            launcher = None
-            if getattr(self, 'HYDRA_LAUNCHER', None) is not None:
-                launcher = [x for x in self.HYDRA_LAUNCHER if x in self.hydra_info.get('launcher', [])]
-                if launcher:
-                    self.log.debug("make_mpiexec_hydra_options: HYDRA: launcher %s, using first one", launcher)
-                else:
-                    self.log.debug("make_mpiexec_hydra_options: no launcher from HYDRA_LAUNCHER %s and hydra_info %s",
-                                   self.HYDRA_LAUNCHER, self.hydra_info)
+            launcher = default_launcher
+
+        if launcher:
+            # filter out the launchers that are not in self.hydra_info
+            avail_launcher = [x for x in launcher if x in self.hydra_info.get('launcher', [])]
+            if avail_launcher:
+                self.log.debug("make_mpiexec_hydra_options: HYDRA: launcher %s, using first one", avail_launcher)
+                self.mpiexec_options.append("-%s %s" % (self.HYDRA_LAUNCHER_NAME, avail_launcher[0]))
 
-            launcher_exec = self.HYDRA_LAUNCHER_EXEC
-            if not launcher:
-                launcher_exec = self.get_rsh()
             else:
-                self.mpiexec_options.append("-%s %s" % (self.HYDRA_LAUNCHER_NAME, launcher[0]))
+                self.log.debug("make_mpiexec_hydra_options: no launcher from HYDRA_LAUNCHER %s and hydra_info %s",
+                                self.HYDRA_LAUNCHER, self.hydra_info)
+
+                launcher_exec = getattr(self, 'HYDRA_LAUNCHER_EXEC', None)
+
+                if launcher_exec is not None:
+                    self.log.debug("make_mpiexec_hydra_options: HYDRA using launcher exec %s", launcher_exec)
+                else:
+                    launcher_exec = self.get_rsh()
```

Comment:
ok, this is probably because of `pbsssh` being the default `launcher_exec`?

you should change that: only use `pbsssh` as default `launcher_exec` if `ssh` is used as a launcher?